plutus-core/plutus-core/test/Evaluation/Spec.hs

```diff
@@ -1,11 +1,247 @@
+{-# LANGUAGE AllowAmbiguousTypes   #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+
 module Evaluation.Spec where
 
+import Control.Monad.Except
+import Data.ByteString qualified as BS
+import Data.Functor ((<&>))
+import Data.Int (Int64)
+import Data.List.Extra qualified as List
+import Data.Text (Text)
+import Data.Type.Equality
+import Data.Typeable (splitTyConApp)
 import Evaluation.Machines (test_machines)
-
+import Hedgehog hiding (Opaque, Var, eval)
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import PlutusCore hiding (Term)
+import PlutusCore qualified as PLC
+import PlutusCore.Builtin
+import PlutusCore.Data (Data (..))
+import PlutusCore.Generators
+import Prettyprinter
 import Test.Tasty
+import Test.Tasty.Hedgehog
+import Type.Reflection
+import Unsafe.Coerce
+
+type Term = PLC.Term TyName Name DefaultUni DefaultFun ()
+
+test_builtinsDon'tThrow :: TestTree
+test_builtinsDon'tThrow =
+  testGroup
+    "Builtins don't throw"
+    . fmap (\fun -> testProperty (show fun) $ prop_builtinsDon'tThrow fun)
+    $ List.enumerate
+      -- TODO: remove this
+      List.\\ [ VerifySignature,
+                VerifyEcdsaSecp256k1Signature,
+                VerifySchnorrSecp256k1Signature
+              ]
+
+prop_builtinsDon'tThrow :: DefaultFun -> Property
+prop_builtinsDon'tThrow bn = property $ do
+  (args, argStrings) <- unzip <$> forAllNoShow (genArgs bn)
+  let (res, logs) = runEmitter . runExceptT $ eval args argStrings
+  case res of
+    Right _ -> success
+    Left err -> do
+      liftIO $ do
+        putStrLn "Builtin function evaluation failed"
+        putStrLn $ "Function: " <> show bn
+        putStrLn $ "Arguments: " <> show argStrings
```

Comment:
Our default rendering goes via the `Pretty` class, not `Show`. And for builtins we had to introduce `PrettyConst` to overload the default behavior of `Pretty`. It should suffice to replace each `show` here with a `displayConst` from `PlutusCore.Pretty.PrettyConst` (the function should be reexported from `PlutusCore.Pretty`, but it seems like we failed at that).